xarray/core/dtypes.py

```diff
@@ -61,7 +61,12 @@ def maybe_promote(dtype: np.dtype) -> tuple[np.dtype, Any]:
     # N.B. these casting rules should match pandas
     dtype_: np.typing.DTypeLike
     fill_value: Any
-    if isdtype(dtype, "real floating"):
+    if np.issubdtype(dtype, np.dtypes.StringDType()):
```

Comment:
`numpy.dtypes` is new in `numpy>=2`
```suggestion
if hasattr(np, "dtypes") and np.issubdtype(dtype, np.dtypes.StringDType()):
```